agent/agent.go

```diff
@@ -164,8 +168,35 @@ func (a *Agent) generateNomadClient() error {
 // reload triggers the reload of sub-routines based on the operator sending a
 // SIGHUP signal to the agent.
 func (a *Agent) reload() {
+	a.logger.Info("reloading Autoscaler configuration")
+
+	// Reload config files from disk.
+	newCfg, err := config.LoadPaths(a.configPaths)
+	if err != nil {
+		a.logger.Error("failed to reload Autoscaler configuration", "error", err)
+		return
```

Comment:
I thought about this, but I am not 100% what would be the best approach. I think the options are:

#### 1. Current approach
I think it's fine, but it does hide errors from operators.

#### 2. Ignore config error and reload sources
I think this would also be fine but still hides the error. 

The only potential probably I see is that we would need to be careful in the future to recognize that, in this state, the configuration being provided to the Autoscaler is invalid and should be ignored, but everyone reads comments right 😄 

#### 3. Just crash
In this state, the Autoscaler wouldn't be able to start, so there could be an argument that it shouldn't be able to restart either. 

This is the most operator-visible option, and quite disruptive. But maybe desirable? Given that the Autoscaler sits in the control plane, it might be better for it to crash sooner and loudly (failed allocs will be displayed in the UI) rather than during an outage caused by a policy not being evaluated?

Reply:
After some discussion we decided to go with option 3 for now. This could be improved in future releases once we have a better mechanism to surface errors.